metacat-common-server/src/main/java/com/netflix/metacat/common/server/converter/TypeConverterProvider.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.netflix.metacat.common.server.converter;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.Maps;
+import com.netflix.metacat.common.MetacatRequestContext;
+import com.netflix.metacat.common.server.Config;
+import com.netflix.metacat.common.server.connectors.ConnectorTypeConverter;
+import com.netflix.metacat.common.util.MetacatContextManager;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import java.util.Map;
+
+/**
+ * Type converter provider.
+ * @author amajumdar
+ * @author tgianos
+ */
+public class TypeConverterProvider implements Provider<ConnectorTypeConverter> {
+    private final Config config;
+    private Map<String, ConnectorTypeConverter> registry = Maps.newHashMap();
+
+    /**
+     * Constructor.
+     * @param config config
+     */
+    @Inject
+    public TypeConverterProvider(
+        final Config config
+    ) {
+        this.config = config;
+    }
+
+    @Override
+    public ConnectorTypeConverter get() {
+        final MetacatRequestContext requestContext = MetacatContextManager.getContext();
```

Comment:
If you use fork of the request to multiple threads, then you will lose the context unless you set it in each of the child threads. It should be okay if you do not fork it.